frontend/src/features/public-form/components/FormPaymentPage/stripe/components/PaymentBlock.tsx

```diff
@@ -161,7 +161,10 @@ export const StripePaymentBlock = ({
     return 'Please make payment.'
   }, [formTitle])
 
-  return form?.responseMode === FormResponseMode.Encrypt ? (
+  if (form?.responseMode !== FormResponseMode.Encrypt) {
+    return <></>
+  }
```

Comment:
In this case, should we be redirecting to some other page / throwing an error instead? Rather than silently return an empty diff. 
Not sure if this is important this decision is important to the scope of this PR though.

Reply:
Ah, I see that returning `<></>` is something we already do originally. Just curious, why did you extract it out? Is it for better readability or something?

Reply:
Generally we should try to leave the happy path all the way to the last. Moving this up will allow the reader to understand that this component will short-circuit if we are not in Encrypt mode.

I found that I should also add `!form` in this comparator.

Reply:
Oh? Do we want to use a non-null assertion here though? The component will completely break if the public form context returns undefined.

Reply:
Yeah. So we can remove the need to continuously check if `form` is falsey. I.e., later in the code there's a `form?.payments_field?.amount_cents`.

Reply:
It is true though that `PaymentBlock` shouldn't need to check for `form`, as it is presumed to be present. I would also prefer to change every component to be less smart...